api/integrations/slack/views.py

```diff
@@ -81,8 +81,8 @@ def slack_oauth_callback(self, request, environment_api_key):
             code, self._get_slack_callback_url(environment_api_key)
         )
 
-        SlackConfiguration.objects.update_or_create(
-            project=env.project, defaults={"api_token": bot_token}
+        SlackConfiguration.objects.all_with_deleted().update_or_create(
```

Comment:
Should we create a test case that covers this?

Reply:
I have fought with this for a long time to write a test that makes sense but, without refactoring all this code into a service layer, there is no easy (not hacky) way to test it. In doing this though, I realised that there is currently no way to delete a SlackConfiguration object anyway so, the way I see, the pragmatic solutions are: 

 1. Keep the change I made _just in case_ it happens in the future, but accept there is no test case to cover it
 2. Just revert this change so we are at least alerted via Sentry if it does happen

Reply:
I think I prefer 2 because it should not happen under normal circumstances

Reply:
Yep, ok, I've done that.